webserver/handlers.go

```diff
@@ -0,0 +1,501 @@
+package webserver
+
+import (
+	"bytes"
+	"challenge/cache"
+	"challenge/middleware"
+	"challenge/models"
+	"challenge/storage"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"time"
+
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/gorilla/mux"
+	"golang.org/x/crypto/bcrypt"
+)
+
+// SigninExistingUser logs in existing user
+func SigninExistingUser(w http.ResponseWriter, r *http.Request) {
+	var existingUser models.User
+	err := json.NewDecoder(r.Body).Decode(&existingUser)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprintf(w, "Error decoding request body")
+		return
+	}
```

Comment:
This part repeats everywhere in all handlers and middleware, how would you approach in improving this? Also this is very prone to `WriteHeader called multiple times` error. Ideas?

Reply:
Of course this code can be replaced by a function that that does the same exact thing. This would not probably address the issue with the error you are referring though. A possible solution would be to handle this functionality, by a middleware that adds it on all requests.